tests/src/Functional/DeveloperSyncTest.php

```diff
@@ -0,0 +1,216 @@
+    }
+
+    // Create users in Drupal.
+    _apigee_edge_set_sync_in_progress(TRUE);
+    for ($i = 0; $i < 1; $i++) {
+      $this->drupalUsers[$i] = $this->createAccount([], TRUE, $this->prefix, FALSE);
+    }
+    _apigee_edge_set_sync_in_progress(FALSE);
+
+    $this->drupalLogin($this->rootUser);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function tearDown() {
+    $remote_ids = array_map(function ($record): string {
+      return $record['email'];
+    }, $this->edgeDevelopers);
+    $drupal_emails = array_map(function (UserInterface $user): string {
+      return $user->getEmail();
+    }, $this->drupalUsers);
+    $ids = array_merge($remote_ids, $drupal_emails);
+    foreach ($ids as $id) {
+      Developer::load($id)->delete();
+    }
+    parent::tearDown();
+  }
+
+  /**
+   * Verifies that the Drupal users and the Edge developers are synchronized.
+   */
+  protected function verify() {
+    $all_users = [];
+    /** @var \Drupal\user\UserInterface $account */
+    foreach (User::loadMultiple() as $account) {
+      $email = $account->getEmail();
+      if ($email && $email !== 'admin@example.com') {
+        $this->assertTrue($this->filter ? (bool) preg_match($this->filter, $email) : TRUE, "Email ({$email}) is filtered properly.");
+        $all_users[$email] = $email;
+      }
+    }
+
+    unset($all_users[$this->rootUser->getEmail()]);
+
+    foreach ($this->edgeDevelopers as $edgeDeveloper) {
+      /** @var \Drupal\user\Entity\User $account */
+      $user = user_load_by_mail($edgeDeveloper->getEmail());
+      $this->assertNotEmpty($user, 'Account found: ' . $edgeDeveloper->getEmail());
+      $this->assertEquals($edgeDeveloper->getUserName(), $user->getAccountName());
+      $this->assertEquals($edgeDeveloper->getFirstName(), $user->get('first_name')->value);
+      $this->assertEquals($edgeDeveloper->getLastName(), $user->get('last_name')->value);
+
+      unset($all_users[$edgeDeveloper->getEmail()]);
+    }
+
+    foreach ($this->drupalUsers as $drupalUser) {
+      /** @var \Drupal\apigee_edge\Entity\DeveloperInterface $developer */
+      $developer = Developer::load($drupalUser->getEmail());
+      $this->assertNotEmpty($developer, 'Developer found on edge.');
+      $this->assertEquals($drupalUser->getAccountName(), $developer->getUserName());
+      $this->assertEquals($drupalUser->get('first_name')->value, $developer->getFirstName());
+      $this->assertEquals($drupalUser->get('last_name')->value, $developer->getLastName());
+
+      unset($all_users[$drupalUser->getEmail()]);
+    }
+
+    $this->assertEquals([], $all_users, 'Only the necessary users were synced. ' . implode(', ', $all_users));
+  }
+
+  /**
+   * Tests Drupal user synchronization.
+   *
+   * @throws \Behat\Mink\Exception\ResponseTextException
+   */
+  public function testUserSync() {
+    $this->drupalGet('/admin/config/apigee-edge/developer-settings/sync');
+    $this->clickLinkProperly(t('Now'));
```

Comment:
Either you or @cnovak will have to update your/his PR. It depends which one gets merged first. :) (The label of the button gets changed in Chris's PR #36)